bake/compose.go

```diff
@@ -208,6 +195,67 @@ func ParseCompose(cfgs []composetypes.ConfigFile, envs map[string]string) (*Conf
 	return &c, nil
 }
 
+func loadComposeFiles(cfgs []composetypes.ConfigFile, envs map[string]string, options ...func(*loader.Options)) (*composetypes.Project, error) {
+	if envs == nil {
+		envs = make(map[string]string)
+	}
+
+	cfgDetails := composetypes.ConfigDetails{
+		ConfigFiles: cfgs,
+		Environment: envs,
+	}
+
+	raw, err := loader.LoadModelWithContext(context.Background(), cfgDetails, append([]func(*loader.Options){func(opts *loader.Options) {
+		projectName := "bake"
+		if v, ok := envs[consts.ComposeProjectName]; ok && v != "" {
+			projectName = v
+		}
+		opts.SetProjectName(projectName, false)
+		opts.SkipNormalization = true
+		opts.SkipValidation = true
+	}}, options...)...)
+	if err != nil {
+		return nil, err
+	}
+
+	filtered := make(map[string]any)
+	for _, key := range []string{"services", "secrets"} {
+		if key == "services" {
+			if services, ok := raw["services"].(map[string]any); ok {
+				filteredServices := make(map[string]any)
+				for svcName, svc := range services {
+					if svc == nil {
+						filteredServices[svcName] = map[string]any{}
+					} else if svcMap, ok := svc.(map[string]any); ok {
+						filteredService := make(map[string]any)
+						for _, svcField := range []string{"image", "build", "environment", "env_file"} {
+							if val, ok := svcMap[svcField]; ok {
+								filteredService[svcField] = val
+							}
+						}
+						filteredServices[svcName] = filteredService
+					}
+				}
+				filtered["services"] = filteredServices
+			}
+		} else if v, ok := raw[key]; ok {
+			filtered[key] = v
+		}
+	}
+
+	if err := composevalidation.Validate(filtered); err != nil {
+		return nil, err
+	}
+
+	return loader.ModelToProject(filtered, loader.ToOptions(&cfgDetails, append([]func(*loader.Options){func(options *loader.Options) {
+		options.SkipNormalization = true
+		options.Profiles = []string{"*"}
```

Comment:
building with all profiles enabled is not consistent with Compose. Is this expected ?

Reply:
We allow all profiles because Bake doesn't support them: https://github.com/docker/buildx/pull/1903. If we add something similar in our spec I think we could reconsider that. cc @colinhemmings

Reply:
In follow-up we could consider mapping compose profiles to bake groups and maybe defining a default profile that would be used if one is defined.